types/htmlbars-inline-precompile/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for htmlbars-inline-precompile 1.0
+// Project: ember-cli-htmlbars-inline-precompile
+// Definitions by: Chris Krycho <https://github.com/chriskrycho>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// This is a bit of a funky one: it's from a [Babel plugin], but is exported for
+// Ember applications as the module `"htmlbars-inline-precompile"`. It acts
+// like a tagged string from the perspective of consumers, but is actually an
+// AST transformation which generates a function as its output. That function in
+// turn [generates a string or array of strings][output] to use with the Ember
+// testing helper `this.render()`.
+//
+// [Babel plugin]: https://github.com/ember-cli/babel-plugin-htmlbars-inline-precompile#babel-plugin-htmlbars-inline-precompile-
+// [output]: https://github.com/emberjs/ember-test-helpers/blob/77f9a53da9d8c19a85b3122788caadbcc59274c2/lib/ember-test-helpers/-legacy-overrides.js#L17-L42
+
+export default function hbs(tagged: TemplateStringsArray): string | string[];
```

Comment:
that is the issue. if you use babel it will be transformed in a manner that would work, if you use TS compiler it will not, and you will get no error. that is why this recommendation is there.

The TS compiler transforms default imports to access to a property named `"default"` on the module. if the module at runtime does not have that property it will fail. since this is a commonjs module it does not have a `"default"` property at run-time.
The TS compiler has a flag to tell it to ignore these errors if the user is going to use a transpiler that handles default imports like babel, or a loader that can synthesize them like systemJs, that option is `--allowSyntheticDefaultImports `.